components/match2/wikis/hearthstone/match_group_input_custom.lua

```diff
@@ -115,38 +119,30 @@ end
 ---@param opponentIndex integer
 ---@return {character: string?, player: string}[]
 function MapFunctions.getTeamParticipants(mapInput, opponent, opponentIndex)
+	local oppPrefix = 'o' .. opponentIndex
+
 	local players = Array.mapIndexes(function(playerIndex)
-		return Logic.nilIfEmpty(mapInput['o' .. opponentIndex .. 'p' .. playerIndex])
+		return Logic.nilIfEmpty(mapInput[oppPrefix .. 'p' .. playerIndex])
+			or Logic.nilIfEmpty(mapInput[oppPrefix .. 'c' .. playerIndex])
 	end)
 
-	local participants, unattachedParticipants = MatchGroupInputUtil.parseParticipants(
+	return MatchGroupInputUtil.parseMapPlayers(
 		opponent.match2players,
 		players,
 		function(playerIndex)
-			local prefix = 'o' .. opponentIndex .. 'p' .. playerIndex
+			local prefix = oppPrefix .. 'p' .. playerIndex
 			return {
 				name = mapInput[prefix],
 				link = Logic.nilIfEmpty(mapInput[prefix .. 'link']),
 			}
 		end,
 		function(playerIndex, playerIdData, playerInputData)
-			local prefix = 'o' .. opponentIndex .. 'p' .. playerIndex
 			return {
 				player = playerIdData.name or playerInputData.link,
-				character = MapFunctions.readCharacter(Logic.nilIfEmpty(mapInput[prefix .. 'char']).character),
+				class = MapFunctions.readCharacter(Logic.nilIfEmpty(mapInput[oppPrefix .. 'c' .. playerIndex])),
 			}
 		end
 	)
-
-	Array.forEach(unattachedParticipants, function(participant)
-		table.insert(opponent.match2players, {
-			name = participant.player,
-			displayname = participant.player,
-		})
-		participants[#opponent.match2players] = participant
-	end)
-
-	return participants
 end
```

Comment:
fwiw this changes functionality a bit
before it extended the opponent.match2players table now it does not
(fine for me if not needed)

Reply:
In the end should be the same no?

Reply:
no it is not
the old stuff extended on match level and map level
the new only on map level
again: if not needed/wanted to also extend on match level fine for me, just pointing out the difference

Reply:
Should be ok